modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/util/CETUtil.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.client.extension.web.internal.util;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.Validator;
+
+/**
+ * @author Binh Tran
+ */
+public class CETUtil {
+
+	public static String replaceNonwordCharacterToUnderline(
```

Comment:
:thinking: Let's see what Brian thinks of this naming. I would have named this method by what it is used for, not upon what it does. IMO, that is better, because we create methods to define abstractions, not to reuse code snippets.

So, instead of `replaceNonwordCharacterToUnderline` something like `convertERCToPortletId`.

Even in the case where we would need to `replaceNonwordCharacterToUnderline` for another thing that wouldn't be `convertERCToPortletId`, I wouldn't name the method `replaceNonwordCharacterToUnderline`, but would create two public methods called:

- `convertERCToPortletId`
- `doTheOtherThing`

And then a private one named `replaceNonwordCharacterToUnderline` so that the other two can delegate on it. This way you define proper abstractions and don't have to duplicate the implementation.